### PR TITLE
Upgrade to ember 1.12

### PR DESCRIPTION
**Pull request by hakubo**
- remove bind-attr everywher
- use new computed property syntax
- to bind to class attribute we have to use inline if now aka {{if isExpanded 'expanded' 'collapsed'}}

Last thing I am seeing now are warnings about XSS in style attribute despite that all input is validated and marked as safe.

### CODE REVIEW

File: front/scripts/main/components/AdSlotComponent.ts [hunk 1]
```diff
@@ -26,8 +26,13 @@ App.AdSlotComponent = Em.Component.extend({
 	 * If getter is accessed before setter (before Ember cache is filled with value)
 	 * the default is false (show ads)
 	 */
-	noAds: Em.computed(function (key?: string, value?: any): boolean {
-		return (arguments.length === 2 && value !== '' && value !== '0');
+	noAds: Em.computed({
+		get() {
```
Comment (rogatty) [inline front/scripts/main/components/AdSlotComponent.ts]: `: boolean`, right? If I'm correct then add it everywhere where missing.

File: front/scripts/main/components/ImageLightboxComponent.ts [hunk 2]
```diff
@@ -81,23 +86,33 @@ App.ImageLightboxComponent = Em.Component.extend(App.ArticleContentMixin, App.Lo
...
 	/**
 	 * @desc calculates Y for panning with respect to maxY
 	 */
-	newY: Em.computed('viewportSize', 'imageHeight', function (key: string, value?: number): number {
-		if (typeof value !== 'undefined' && this.get('imageHeight') > this.get('viewportSize').height) {
-			return this.limit(value, this.get('maxY'));
-		}
+	newY: Em.computed('viewportSize', 'imageHeight', {
+		get() {
+			return 0;
```
Comment (kenkouot) [inline front/scripts/main/components/ImageLightboxComponent.ts]: The getter always returns 0?

File: front/scripts/main/components/ImageLightboxComponent.ts [hunk 3]
```diff
@@ -41,12 +43,15 @@ App.ImageLightboxComponent = Em.Component.extend(App.ArticleContentMixin, App.Lo
...
-		if (value >= 1) {
-			return Math.min(this.maxZoom, value);
+	scale: Em.computed({
+		get() {
+			return 1;
+		},
+		set(key: string, value: number) {
+			if (value >= 1) {
+				return Math.min(this.maxZoom, value);
+			}
```
Comment (rogatty) [inline front/scripts/main/components/ImageLightboxComponent.ts]: `else` is missing.

**Comment by kenkouot** [comment]
Looking good @hakubo! Can you make sure to update the guidelines as needed? https://github.com/Wikia/guidelines/blob/master/Ember/README.md

File: front/scripts/main/components/ImageLightboxComponent.ts [hunk 4]
```diff
@@ -81,23 +86,33 @@ App.ImageLightboxComponent = Em.Component.extend(App.ArticleContentMixin, App.Lo
...
-	newX: Em.computed('viewportSize', 'imageWidth', function (key: string, value?: number): number {
-		if (typeof value !== 'undefined' && this.get('imageWidth') > this.get('viewportSize').width) {
-			return this.limit(value, this.get('maxX'));
-		}
+	newX: Em.computed('viewportSize', 'imageWidth', {
+		get() {
+			return 0;
+		},
+		set (key: string, value: string) {
+			if (this.get('imageWidth') > this.get('viewportSize').width) {
```
Comment (rogatty) [inline front/scripts/main/components/ImageLightboxComponent.ts]: `this.get('viewportSize.width')` seems safer.

File: front/scripts/main/components/ImageLightboxComponent.ts (hunk 2 above)
Comment (hakubo) [inline front/scripts/main/components/ImageLightboxComponent.ts]: Remember that get is cached
So this is just setting default value

File: front/templates/main/components/article-comments.hbs [hunk 5]
```diff
@@ -1,17 +1,17 @@
...
+	<div {{action 'toggleComments'}} class='show-comments-btn page-btn {{if isCollapsed 'collapsed'}}'>
 		{{i18n 'app.article-comments-label' count='commentsCount'}}
 		{{svg 'chevron' viewBox='0 0 12 7' class='icon chevron'}}
 	</div>
 {{/if}}
 {{#if showComments}}
 	{{#if model.comments}}
-		<button {{bind-attr class=':page-btn prevButtonShown::hidden'}} {{action 'prevPage'}}>{{i18n 'app.comments-prev-button'}}</button>
-		<ul {{bind-attr class=':comments commentsHidden:hidden'}}>
+		<button class='page-btn {{if prevButtonShown '' 'hidden'}}' {{action 'prevPage'}}>{{i18n 'app.comments-prev-button'}}</button>
```
Comment (rogatty) [inline front/templates/main/components/article-comments.hbs]: Would `{{if !prevButtonShown 'hidden'}}` work?

Comment (hakubo) [inline front/templates/main/components/article-comments.hbs]: good question :)
don't know if they support such case, will have a look

**Comment by hakubo** [comment]
@kenkouot sure, will do!

**Comment by rogatty** [comment]
LGTM

**Comment by kenkouot** [comment]
LGTM!

**Comment by vforge** [comment]
LGTM! :ship: it

**Pull request merged**